adafruit_matrixportal/fakerequests.py

```diff
@@ -0,0 +1,44 @@
+# SPDX-FileCopyrightText: 2020 Melissa LeBlanc-Williams, written for Adafruit Industries
+#
+# SPDX-License-Identifier: Unlicense
+"""
+`adafruit_matrixportal.fakerequests`
+================================================================================
+
+Helper library for the Adafruit RGB Matrix Shield + Metro M4 Airlift Lite.
+
+* Author(s): Melissa LeBlanc-Williams
+
+Implementation Notes
+--------------------
+
+**Hardware:**
+
+* `Adafruit Metro M4 Express AirLift <https://www.adafruit.com/product/4000>`_
+* `Adafruit RGB Matrix Shield <https://www.adafruit.com/product/2601>`_
+* `64x32 RGB LED Matrix <https://www.adafruit.com/product/2278>`_
+
+**Software and Dependencies:**
+
+* Adafruit CircuitPython firmware for the supported boards:
+  https://github.com/adafruit/circuitpython/releases
+
+"""
+
+__version__ = "0.0.0-auto.0"
+__repo__ = "https://github.com/adafruit/Adafruit_CircuitPython_MatrixPortal.git"
+
+
+class Fake_Requests:
+    """For faking 'requests' using a local file instead of the network."""
+
+    def __init__(self, filename):
+        self._filename = filename
+        with open(filename, "r") as file:
+            self.text = file.read()
+
+    def json(self):
+        """json parsed version for local requests."""
+        import json  # pylint: disable=import-outside-toplevel
+
+        return json.loads(self.text)
```

Comment:
Use json.load here and open the file here as well. That way it never loads the full string into memory.

Reply:
Although this suggestion will improve the `.json()` function, I think this will break trying to access the .text property unless json() is called first. Would you like me to create a text property that loads and returns the text of the file as well?